relationalai-sdk-snowflake/relationalai/snowflake_sdk/api.py

```diff
@@ -0,0 +1,269 @@
+    """)
+
+
+def use_database(session: Session, database: str) -> List[Row]:
+    res = session.sql(f"call USE_RAI_DATABASE('{database}')").collect()
+
+    if res[0][0] != database:
+        rsp = json.loads(res[0][0])
+
+        if not rsp["success"]:
+            raise Exception(rsp["message"])
+
+    return res
+
+
+def get_current_database(session: Session) -> DataFrame:
+    return session.sql("select CURRENT_RAI_DATABASE() as current_database")
+
+
+#################################
+# ENGINE
+#################################
+
+def create_engine(session: Session, engine: str, size: str = 'XS') -> List[Row]:
+    return session.sql(f"select CREATE_RAI_ENGINE('{engine}', '{size}') as status").collect()
+
+
+def delete_engine(session: Session, engine: str) -> List[Row]:
+    return session.sql(f"select DELETE_RAI_ENGINE('{engine}') as status").collect()
+
+
+def get_engine(session: Session, engine: str) -> DataFrame:
+    return session.sql(f"""
+        select
+            res:id::string           as id,
+            res:account_name::string as account_name,
+            res:created_by::string   as created_by,
+            res:created_on::string   as created_on,
+            res:name::string         as name,
+            res:region::string       as region,
+            res:size::string         as size,
+            res:state::string        as state
+        from
+            (select GET_RAI_ENGINE('{engine}') as res);
+    """)
+
+
+def list_engines(session: Session) -> DataFrame:
+    return session.sql(f"""
+        select
+            value:id::string           as id,
+            value:account_name::string as account_name,
+            value:created_by::string   as created_by,
+            value:created_on::string   as created_on,
+            value:name::string         as name,
+            value:region::string       as region,
+            value:size::string         as size,
+            value:state::string        as state
+        from
+            (select LIST_RAI_ENGINES() as res),
+            lateral flatten (input => res);
+""")
+
+
+def use_engine(session: Session, engine: str) -> List[Row]:
+    res = session.sql(f"CALL USE_RAI_ENGINE('{engine}')").collect()
+
+    if res[0][0] != engine:
+        rsp = json.loads(res[0][0])
+
+        if not rsp["success"]:
+            raise Exception(rsp["message"])
+
+    return res
+
+
+def get_current_engine(session: Session) -> DataFrame:
+    return session.sql("select CURRENT_RAI_ENGINE() as current_engine")
+
+
+#################################
+# Transaction
+#################################
+
+
+def exec(session: Session, database: str, engine: str, query: str, data=None, readonly: bool = True) -> DataFrame:
+    return session.sql(f"select EXEC('{database}', '{engine}', '{query}', {data if data else 'null'}, {readonly})")
+
+
+def exec_into(session: Session, database: str, engine: str, query: str, warehouse: str, target: str, data=None, readonly: bool = True) -> DataFrame:
```

Comment:
Rationalizing results of "do something" being returned as List[Row] and "get me data" returned as DataFrame, exec_into leans on the "do something" side for me. The results simply say "ok"

Reply:
Right, my miss. Will fix that.

Reply:
Fixed